### ISSUE TITLE
Simple Network Builder gives all 7000 predicted sample outputs identical.

### ISSUE THREAD

**Issue by Deewrat** [Microsoft/CNTK#1684]
I am new at CNTK and trying to develop a command line configuration file to read a data file of about 7000 samples (each with 5 input features and 1 output label), all floating point numbers. I use the SimpleNetworkBuilder. After training (no matter how many epochs) all the predicted labels are identical (to 5 decimal points) despite different input features. What am I doing wrong?

ConfigFile is:
--------------------------------------------------------------------------------------------------------------------------------------
# simple recurrent network via the configfile route, not python
	rootDir = "c:\cntk_working\simpleRecurr"
	configDir = "$rootDir$"
	dataDir =  "$rootDir$\Data"
	outputDir = "$rootDir$\Output"
	modelDir = "$rootDir$\Models"

	command = simpleRecurrTrain:simpleRecurrTest:simpleRecurrWrite

	simpleRecurrTrain = [
		action = "train"
		modelPath = "$modelDir$\Models"
		
		#network definition

		simpleNetworkBuilder = [
			layerSizes	  = 5:10:10:1
			trainingCriterion  = "SquareError"
			evalCriterion	  = "SquareError"
			layerTypes 	  = "Sigmoid"
			applyMeanVarNorm  = true
			rnnType 	  = "LSTM"
			recurrentLayer    = 2:3
		]

		#SGD learner config

		SGD = [
			modelPath 		= "$modelDir$\models"
			epochSize 		= 0 #use it all
			minibatchSize 		= 128
			maxEpochs               = 500
			autoLearnRate = true
			learningRatesPerSample = 0.01
		]

		#reader config

		reader = [
			readerType = "CNTKTextFormatReader"
			verbosity = 0
			file = "$dataDir$\simpleRecurr.dat"
			input = [
				features = [
					dim = 5 # only 1 kind of feature
					format = "dense"
					   ]
				labels = [
					dim = 1 # only 1 kind of label
					format  "dense"
					]
				]			
		]
	]

	# test

	simpleRecurrTest [

	    action = "test"
	    modelPath = "$modelDir$\models"

	    reader = [
        	readerType = "CNTKTextFormatReader"
        	file = "$dataDir$\simpleRecurr.tst"  # test data
			input = [
				features = [
					dim = 5 # only 1 kind of feature
					format = "dense"
					   ]
				labels = [
					dim = 1 # only 1 kind of label
					format  "dense"
					]
			]
		 ]       	    	
	]					

	simpleRecurrWrite [      # output the results

	    action = "write"
	    modelPath = "$modelDir$\models"

	    reader = [
        	readerType = "CNTKTextFormatReader"
        	file = "$dataDir$\simpleRecurr.tst"  # test data, or do we want the learning data too?
			input = [
				features = [
					dim = 5 # only 1 kind of feature
					format = "dense"
					   ]
				labels = [
					dim = 1 # only 1 kind of label
					format  "dense"
					]
			]
		]        	
    	
    		outputPath = "$outputDir$\write_stuff"  # dump the output to this text file
	]					
	
*****************************************************************************************************************

**Comment by PeterGriffioen** [comment]
If using CPU then I had a similar problem a month ago before RC2. On installing RC2 I found that I had been pointing at older versions of CNTK libraries for some time. So as an easy check, make sure you have only the latest version of CNTK libraries on your machine. Also have you run the MSMpiSetup.exe in the prerequisites directory? Just a thought.

**Comment by n17s** [comment]
This can happen if your network is not trained properly e.g. a large learning rate, which can cause large weights, which can cause e.g. all non-linearities to saturate regardless of the inputs. 

Feel free to reopen this issue if you have further questions.
